src/lib/mina.ts

```diff
@@ -689,57 +705,68 @@ LocalBlockchain satisfies (...args: any) => Mina;
  * Represents the Mina blockchain running on a real network
  */
 function Network(graphqlEndpoint: string): Mina;
-function Network(endpoints: {
+function Network(options: {
+  networkId?: NetworkId;
   mina: string | string[];
   archive?: string | string[];
   lightnetAccountManager?: string;
 }): Mina;
 function Network(
-  input:
+  options:
     | {
+        networkId?: NetworkId;
         mina: string | string[];
         archive?: string | string[];
         lightnetAccountManager?: string;
       }
     | string
 ): Mina {
+  let minaNetworkId: NetworkId = 'testnet';
   let minaGraphqlEndpoint: string;
   let archiveEndpoint: string;
   let lightnetAccountManagerEndpoint: string;
 
-  if (input && typeof input === 'string') {
-    minaGraphqlEndpoint = input;
+  if (options && typeof options === 'string') {
+    minaGraphqlEndpoint = options;
     Fetch.setGraphqlEndpoint(minaGraphqlEndpoint);
-  } else if (input && typeof input === 'object') {
-    if (!input.mina)
+  } else if (options && typeof options === 'object') {
+    if (options.networkId) {
+      if (options.networkId !== 'mainnet' && options.networkId !== 'testnet') {
+        throw Error(
+          "Invalid network id specified. Please use 'mainnet' or 'testnet'"
+        );
+      }
```

Comment:
remove this check

Reply:
https://github.com/o1-labs/o1js/pull/1387#discussion_r1469463590